### PR TITLE
Fix `Runtime#unsafe.fork` and `Runtime#unsafe.runToFuture`

### LINKED ISSUE #8751: `Runtime#unsafe.fork` doesn't fork synchronous effects
Given the reproduction below, I'm expecting `second` to be printed first in the console. However, the following program blocks on the first effect, printing:

```
first
second
finished
```

I believe the issue is on [this line](https://github.com/zio/zio/blob/4a298040e79b18f8a2a3bc846ead8383a95ea5f8/core/shared/src/main/scala/zio/Runtime.scala#L127), as `start` won't fork the effect until an async operation takes place. The call to `start` should be replaced with `startConcurrently` in order for the fiber to be forked immediately.

Note that [`runToFuture` also uses `start`](https://github.com/zio/zio/blob/4a298040e79b18f8a2a3bc846ead8383a95ea5f8/core/shared/src/main/scala/zio/Runtime.scala#L178), and I think it should probably be changed

### Repro

```scala
import zio.*

def fork(f: UIO[Unit]) = Unsafe.unsafely(Runtime.default.unsafe.fork(f))

val f1 = ZIO.succeed {
  Thread.sleep(1000)
  println("first")
}

val f2 = ZIO.succeed {
  println("second")
}

@main def run = {
  fork(f1)
  fork(f2)
  Thread.sleep(2000)
  println("finished")
}
```

### PR DESCRIPTION
**Pull request by kyri-petrou**
/fixes #8751

### CODE REVIEW

**Comment by ghostdogpr** [comment]
How about `cancelFiber.start` in the future case?

**Comment by kyri-petrou** [comment]
> How about `cancelFiber.start` in the future case?

`interruptAs` only adds a message to the running fiber's queue and then awaits asynchronously. Therefore I think we can get away with using `start` here and have the submission run on the calling thread

**Review by ghostdogpr — DISMISSED** [review]

**Comment by guizmaii** [comment]
Would it be possible to add a test maybe? 🤔

**Comment by kyri-petrou** [comment]
> Would it be possible to add a test maybe? 🤔

Done 👍

**Comment by kyri-petrou** [comment]
Hmm is there anything that cross-compiles to ScalaJS that is equivalent of `Thread.sleep`? Otherwise I can just move the test to the JVM suite

File: core-tests/shared/src/test/scala/zio/RuntimeSpec.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        val fut2     = rt.runToFuture(f2)
+        ZIO.fromFuture(_ => fut1) *> ZIO.fromFuture(_ => fut2) *> assertCompletes
+      }
+    }
+  ) @@ TestAspect.timeout(5.seconds)
+
+  private def mkEffects() = {
+    val loop = new AtomicBoolean(true)
+
+    val f1 = ZIO.succeed {
```
Comment (jdegoes) [inline core-tests/shared/src/test/scala/zio/RuntimeSpec.scala]: Better to do something like `(ZIO.succeed(loop.get()) <* ZIO.yieldNow).retryUntil(_ == true)` so it works on JS too and will be more efficient.

Comment (kyri-petrou) [reply core-tests/shared/src/test/scala/zio/RuntimeSpec.scala]: We can't use `ZIO.yieldNow`, as `.start` will fork the fiber in that case. It needs to be a fully synchronous effect

Comment (jdegoes) [reply core-tests/shared/src/test/scala/zio/RuntimeSpec.scala]: Then you can do `@@ jvmOnly` or equivalent to run it only on the JVM.

Comment (kyri-petrou) [reply core-tests/shared/src/test/scala/zio/RuntimeSpec.scala]: Agh I tried that but turns out it fails to compile with Scala native due to `Thread.sleep`. I moved the test suite under `/jvm` instead

**Review by hearnadam — APPROVED** [review]

**Review by ghostdogpr — APPROVED** [review]

**Review by jdegoes — APPROVED** [review]

**Pull request merged**